### ISSUE TITLE
ClassCastException while trying to get eval response of Lua script that returns a list

### ISSUE THREAD

**Issue by hcadavid** [xetorthio/jedis#1504]
### Expected behavior

Original REDIS-Lua script that returns and clears a list when the fourth element is added: 

`eval "local res;if (redis.call('RPUSH','dalist',ARGV[1])==4) then res=redis.call('LRANGE','dalist',0,3);redis.call('DEL','dalist');return res; else return {} end" 1 argname 9999
`
Jedis code that invokes the previous script inside a transaction:

```
Jedis jedis = JedisUtil.getPool().getResource();
jedis.watch("dalist");
Transaction tx = jedis.multi();
Response<Long> r=tx.rpush("dalist","2322");
tx.eval("local res;if (redis.call('RPUSH','dalist',ARGV[1])==4) then res=redis.call('LRANGE','dalist',0,3);redis.call('DEL','dalist');return res; else return {} end",0,"99999");
        
List<Object> rs=tx.exec();
```

Expected: an empty array/4-elements array

### Actual behavior

Exception in thread "main" java.lang.ClassCastException: java.util.ArrayList cannot be cast to [B
	at redis.clients.jedis.BuilderFactory$5.build(BuilderFactory.java:58)
	at redis.clients.jedis.BuilderFactory$5.build(BuilderFactory.java:56)
	at redis.clients.jedis.Response.build(Response.java:61)
	at redis.clients.jedis.Response.get(Response.java:37)
	at redis.clients.jedis.Transaction.exec(Transaction.java:54)

When exec is invoked:
List<Object> rs=tx.exec();

### Steps to reproduce:

Please create a reproducible case of your problem. Make sure
that case repeats consistently and it's not random
* Just start a Redis instance and run the code above.

### Redis / Jedis Configuration

redis.pool.maxActive=1024
redis.pool.maxIdle=200
redis.pool.maxWait=1000
redis.pool.testOnBorrow=true
redis.pool.testOnReturn=true
redis.ip=127.0.0.1
redis.port=6379

**Comment by hcadavid** [comment]
I tried a simpler example (based on 'jedis' documentation), where an 'eval' is performed within a transaction, and the same problem occurs. Even with the script "return 5", the ClassClassException is thrown:

        Jedis jedis = JedisUtil.getPool().getResource();
        
        Transaction tx = jedis.multi();
        
        String script = "return {KEYS[1],KEYS[2],ARGV[1],ARGV[2],ARGV[3]}";
        List<String> keys = new ArrayList<String>();
        keys.add("key1");
        keys.add("key2");

        List<String> args = new ArrayList<String>();
        args.add("first");
        args.add("second");
        args.add("third");

        tx.eval(script, keys, args);

        tx.exec();

**Comment by marcosnils** [comment]
@hcadavid thx for reporting the issue. Just reproduced it in the 2.9 branch but somehow it works in `master`. Seems like something wasn't downmerged correctly. I'll try to take a look

**Comment by marcosnils** [comment]
Found the problem. Check https://github.com/xetorthio/jedis/pull/607 and https://github.com/xetorthio/jedis/pull/1361. 

It's probable that that change won't be backported to 2.x due to backwards compatibility issues, so you have two options:

1. Use the binary `eval` signature which parses the response correctly: 
```java
  Jedis jedis = new Jedis();
        jedis.watch("dalist");
        Transaction tx = jedis.multi();
        Response<Long> r=tx.rpush("dalist", "2322");
        tx.eval("local res;if (redis.call('RPUSH','dalist',ARGV[1])==4) then res=redis.call('LRANGE','dalist',0,3);redis.call('DEL','dalist');return res; else return {} end".getBytes(), 0, "99999".getBytes());

        List<Object> rs=tx.exec();
```

2. Use master branch. 

I'd recommend the 1st if you don't want to deal with building and pushing a artifact yourself.

**Comment by sazzad16** [comment]
Resolved by #1667
